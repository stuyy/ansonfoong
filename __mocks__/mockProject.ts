import { Project } from "../src/utils/types";

export const mockProject: Project = {
  title: "Dogwrench",
  description:
    "Dogwrench is an on-going project started in late 2020. The idea originated from a simple Shop Management system for a client that owns a car repair shop, but then evolved into an enterprise application for customers and shop owners. Dogwrench aims to connect customers with shop owners and promote automotive services with discounts.",
  image: "/projects/dogwrench.png",
  pathname: "dogwrench",
  technologies: ["React", "Angular", "Nest.js", "MySQL"],
  website: "https://dogwrench.com",
};
