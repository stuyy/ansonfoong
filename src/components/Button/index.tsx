import styles from "./button.module.scss";
import { ButtonHTMLAttributes, DetailedHTMLProps } from "react";
import { ButtonVariant, Size } from "../../utils/types";

export type ButtonProps = DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
> & {
  label: string;
  variant?: ButtonVariant;
  size?: Size;
};

export const Button = ({ label, variant, size, ...props }: ButtonProps) => {
  const classes = `${styles.button} ${
    styles[`button--${variant || "primary"}`]
  } ${styles[`button--${size || "sm"}`]}`;
  return (
    <button {...props} className={classes}>
      {label}
    </button>
  );
};
