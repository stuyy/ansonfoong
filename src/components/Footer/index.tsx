import styles from "./Footer.module.scss";
import {
  FaDiscord,
  FaYoutubeSquare,
  FaLinkedin,
  FaGithubSquare,
} from "react-icons/fa";

export const Footer = () => (
  <footer className={styles.footer}>
    <FaDiscord size={40} />
    <FaYoutubeSquare size={40} />
    <FaLinkedin size={40} />
    <FaGithubSquare size={40} />
  </footer>
);
