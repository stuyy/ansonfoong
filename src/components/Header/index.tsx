import styles from "./index.module.scss";
import { routes } from "../../utils/routes";

export const Header = () => (
  <header className={styles.header}>
    <div>
      <h1>ansonfoong</h1>
    </div>
    <ul>
      {routes.map((route) => (
        <li key={route.name} className={styles.header_link}>
          {route.name}
        </li>
      ))}
    </ul>
  </header>
);
