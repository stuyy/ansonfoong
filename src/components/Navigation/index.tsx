import Link from "next/link";
import styles from "./index.module.scss";
import { routes } from "../../utils/routes";

export const Navigation = () => (
  <header className={styles.header}>
    <div>
      <h1>ansonfoong</h1>
    </div>
    <ul>
      {routes.map((route) => (
        <li key={route.name} className={styles.header_link}>
          <Link href={route.path}>
            <a>{route.name}</a>
          </Link>
        </li>
      ))}
    </ul>
  </header>
);
