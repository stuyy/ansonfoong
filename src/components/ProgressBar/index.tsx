import styles from "./ProgressBar.module.scss";

export type ProgressBarProps = {
  spacing?: string;
  color: string;
  progress: number;
  animate: boolean;
};

export const ProgressBar = ({
  spacing,
  color,
  progress,
  animate,
}: ProgressBarProps) => (
  <div
    className={styles["progress-bar-container"]}
    style={{
      marginLeft: spacing,
    }}
  >
    <div
      className={styles["progress-bar-indicator"]}
      style={{
        width: animate ? `${progress}%` : "0%",
        backgroundColor: color,
      }}
    ></div>
  </div>
);
