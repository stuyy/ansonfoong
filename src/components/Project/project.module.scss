.project-container {
  width: 100%;
  background-color: #232323;
  border-radius: 10px;
  padding: 35px;
  display: flex;
  justify-content: space-between;
  & img {
    width: 175px;
    border-radius: 10px;
  }
}

.project-container-body {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  margin-right: 20px;
}

.project-container-content {
  width: 100%;
  & h2 {
    font-weight: normal;
  }
  & p {
    margin-top: 10px;
    font-size: 14px;
  }
}

.project-container-title {
}

.project-container-description {
}

.project-container-buttons {
}
