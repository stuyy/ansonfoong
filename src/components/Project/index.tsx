import styles from "./project.module.scss";
import { Button } from "../Button";
import { useRouter } from "next/dist/client/router";

export type ProjectComponentProps = {
  title: string;
  description: string;
  src: string;
  reverse?: boolean;
  pathname: string;
};

export const Project = ({
  title,
  description,
  src,
  reverse,
  pathname,
}: ProjectComponentProps) => {
  const router = useRouter();
  const containerStyles = `${styles["project-container"]} ${
    reverse && styles["project-container-reverse"]
  }`;

  const containerBodyStyles = `${styles["project-container-body"]} ${
    reverse && styles["container-body-reverse"]
  }`;

  return (
    <div className={containerStyles}>
      <div className={containerBodyStyles}>
        <div className={styles["project-container-content"]}>
          <h2>{title}</h2>
          <p>{description}</p>
        </div>
        <div>
          <Button
            label="GitHub"
            variant="secondary"
            onClick={() => window.location.assign("http://github.com")}
            style={{ marginRight: "10px" }}
          />
          <Button
            label="View"
            onClick={() => router.push(`/projects/${pathname}`)}
          ></Button>
        </div>
      </div>
      <div>
        <img src={src} alt={src} />
      </div>
    </div>
  );
};
