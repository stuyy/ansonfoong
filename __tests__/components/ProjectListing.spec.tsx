/**
 * @jest-environment jsdom
 */

import { render, RenderResult } from "@testing-library/react";
import { ProjectListing } from "../../src/components/Project";
import { mockProject } from "../../__mocks__/mockProject";

describe("ProjectListing", () => {
  let component: RenderResult;
  beforeEach(() => {
    component = render(<ProjectListing project={mockProject} index={0} />);
  });

  it("should match snapshot", () => expect(component).toMatchSnapshot());
});
