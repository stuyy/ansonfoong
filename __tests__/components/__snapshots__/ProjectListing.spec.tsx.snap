// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectListing should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="project-container project-container-reverse"
      >
        <div
          class="project-container-body container-body-reverse"
        >
          <div
            class="project-container-content"
          >
            <h2>
              Dogwrench
            </h2>
            <p>
              Dogwrench is an on-going project started in late 2020. The idea originated from a simple Shop Management system for a client that owns a car repair shop, but then evolved into an enterprise application for customers and shop owners. Dogwrench aims to connect customers with shop owners and promote automotive services with discounts.
            </p>
          </div>
          <div
            class="project-buttons"
          >
            <button
              class="button button--secondary button--sm"
              style="margin-right: 10px;"
            >
              GitHub
            </button>
            <button
              class="button button--primary button--sm"
            >
              View
            </button>
          </div>
        </div>
        <div>
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTc1IiBoZWlnaHQ9IjE3NSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="/projects/dogwrench.png"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="project-container project-container-reverse"
    >
      <div
        class="project-container-body container-body-reverse"
      >
        <div
          class="project-container-content"
        >
          <h2>
            Dogwrench
          </h2>
          <p>
            Dogwrench is an on-going project started in late 2020. The idea originated from a simple Shop Management system for a client that owns a car repair shop, but then evolved into an enterprise application for customers and shop owners. Dogwrench aims to connect customers with shop owners and promote automotive services with discounts.
          </p>
        </div>
        <div
          class="project-buttons"
        >
          <button
            class="button button--secondary button--sm"
            style="margin-right: 10px;"
          >
            GitHub
          </button>
          <button
            class="button button--primary button--sm"
          >
            View
          </button>
        </div>
      </div>
      <div>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTc1IiBoZWlnaHQ9IjE3NSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="/projects/dogwrench.png"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
